src/LCRmeter-Metrohm_AutolabPGSTAT/main.py

```diff
@@ -0,0 +1,333 @@
+# This Device Class is published under the terms of the MIT License.
+# Required Third Party Libraries, which are included in the Device Class
+# package for convenience purposes, may have a different license. You can
+# find those in the corresponding folders or contact the maintainer.
+#
+# MIT License
+#
+# Copyright (c) 2024 SweepMe! GmbH (sweep-me.net)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+
+# SweepMe! driver
+# * Module: LCRmeter
+# * Instrument: Metrohm Autolab PGSTAT
+
+from __future__ import annotations
+
+import time
+from pathlib import Path
+
+import clr
+from pysweepme import debug
+from pysweepme.EmptyDeviceClass import EmptyDevice
+
+autolab_sdk_path = r"C:\Program Files\Metrohm Autolab\Autolab SDK 2.1\EcoChemie.Autolab.Sdk.dll"
+extension_path = r"C:\Program Files\Metrohm Autolab\Autolab SDK 2.1\EcoChemie.Autolab.SDK.Extensions.dll"
+
+import_failed = False
+try:
+    clr.AddReference(autolab_sdk_path)
+    clr.AddReference(extension_path)
+    from EcoChemie.Autolab.Sdk import Instrument
+    from EcoChemie.Autolab.SDK.Extensions import InstrumentExtensions
+
+except:
+    # TODO: Handle import error
+    import_failed = True
+
+
+class Device(EmptyDevice):
+    """Device class for the Metrohm Autolab PGSTAT used as LCRmeter."""
+
+    description = """
+    <h3>Metrohm Autolab PGSTAT</h3>
+    <p>This driver controls Metrohm Autolab Potentiostats.</p>
+    <h4>Setup</h4>
+    <ul>
+    <li>Install the Autolab SDK 2.1.</li>
+    <li>Create a device configuration of your pump using QmixElements of the CETONI SDK.</li>
+    <li>Copy the LCRmeter_Metrohm-AutolabPGSTAT.ini file to public documents/SweepMe!/CustomFiles. In this file:
```

Comment:
Isn't there a convenience function for this by right-clicking on the version in the version manager or something like that?